internal/net/dialer_test.go

```diff
@@ -1759,69 +1759,162 @@ func Test_dialer_tlsHandshake(t *testing.T) {
+					dialerDualStack: false,
+					addrs:           sync.Map{},
+					der:             &net.Dialer{},
+					dialer:          nil,
+				},
+				checkFunc: func(w want, c *tls.Conn, e error) error {
+					if !c.ConnectionState().HandshakeComplete {
+						return errors.New("Handshake not completed")
+					}
+					return nil
+				},
+				afterFunc: func(a args) {
+					srv.Close()
+					conn.Close()
+					cancel()
+				},
+			}
+		}(),
+		func() test {
+			ctx, cancel := context.WithCancel(context.Background())
+
+			h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(200)
+			})
+			srv := httptest.NewTLSServer(h)
+			srv.TLS.InsecureSkipVerify = true
+
+			d, err := NewDialer()
+			if err != nil {
+				t.Error(err)
+			}
+
+			host, port, _ := SplitHostPort(strings.TrimPrefix(strings.TrimPrefix(srv.URL, "https://"), "http://"))
+			addr := host + ":" + strconv.FormatUint(uint64(port), 10)
+
+			conn, err := d.DialContext(ctx, "tcp", addr)
+			if err != nil {
+				t.Error(err)
+			}
+
+			return test{
+				name: "return error when handshake timeout",
+				args: args{
+					ctx:  ctx,
+					conn: conn,
+					addr: srv.URL,
+				},
+				fields: fields{
+					dnsCache:        false,
+					dnsCachedOnce:   sync.Once{},
+					tlsConfig:       srv.TLS,
+					dialerDualStack: false,
+					addrs:           sync.Map{},
+					der: &net.Dialer{
+						Timeout: 1,
+					},
+					dialer: nil,
+				},
+				checkFunc: func(w want, c *tls.Conn, e error) error {
+					if e == nil {
+						return errors.New("timeout error should be returned")
+					}
+					return nil
+				},
+				afterFunc: func(a args) {
+					srv.Close()
+					conn.Close()
+					cancel()
+				},
+			}
+		}(),
+		func() test {
+			ctx, cancel := context.WithCancel(context.Background())
+
+			h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(200)
+			})
+			srv := httptest.NewTLSServer(h)
+			srv.TLS.InsecureSkipVerify = true
+
+			d, err := NewDialer()
+			if err != nil {
+				t.Error(err)
+			}
+
+			host, port, _ := SplitHostPort(strings.TrimPrefix(strings.TrimPrefix(srv.URL, "https://"), "http://"))
+			addr := host + ":" + strconv.FormatUint(uint64(port), 10)
+
+			conn, err := d.DialContext(ctx, "tcp", addr)
+			if err != nil {
+				t.Error(err)
+			}
+
+			srv.Close()
+
+			return test{
+				name: "return error when host not found",
+				args: args{
+					ctx:  ctx,
+					conn: conn,
+					addr: srv.URL,
+				},
+				fields: fields{
+					dnsCache:        false,
+					dnsCachedOnce:   sync.Once{},
+					tlsConfig:       &tls.Config{},
+					dialerDualStack: false,
+					addrs:           sync.Map{},
+					der:             &net.Dialer{},
+				},
+				checkFunc: func(w want, c *tls.Conn, e error) error {
+					if e == nil {
+						return errors.New("Handshake not completed")
+					}
+					return nil
+				},
+				afterFunc: func(a args) {
+					// srv.Close()
+					conn.Close()
+					cancel()
+				},
+			}
+		}(),
```

Comment:
could you implement without an initial struct for test fields?

Reply:
fixed. can you please review it? thank you!